src/backend/distributed/commands/foreign_constraint.c

```diff
@@ -33,68 +35,45 @@
 #include "utils/ruleutils.h"
 #include "utils/syscache.h"
 
+/*
+ * Flags that can be passed to GetTableForeignConstraintOidsInternal to indicate
+ * which foreign key constraint OIDs are to be extracted
+ */
+#define EXTRACT_REFERENCING_CONSTRAINTS 0x0001
+#define EXTRACT_REFERENCED_CONSTRAINTS 0x0010
+#define EXCLUDE_SELF_REFERENCES 0x0100
+
 /* Local functions forward declarations */
-static bool HeapTupleOfForeignConstraintIncludesColumn(HeapTuple heapTuple, Oid
-													   relationId, int pgConstraintKey,
+static bool HeapTupleOfForeignConstraintIncludesColumn(HeapTuple heapTuple,
+													   Oid relationId,
+													   int pgConstraintKey,
 													   char *columnName);
+static Oid FindForeignKeyOidWithName(List *foreignKeyOids, const
+									 char *inputConstraintName);
 static void ForeignConstraintFindDistKeys(HeapTuple pgConstraintTuple,
 										  Var *referencingDistColumn,
 										  Var *referencedDistColumn,
 										  int *referencingAttrIndex,
 										  int *referencedAttrIndex);
+static List * GetTableForeignConstraintCommandsInternal(Oid relationOid,
+														int flags);
 static Oid get_relation_constraint_oid_compat(HeapTuple heapTuple);
+static List * GetForeignKeyOidsToReferenceTables(Oid relationOid);
```

Comment:
Now that we are using flags, can we have a single method (only GetTableForeignConstraintOids) instead of `GetForeignKeyOidsToReferenceTables` and `GetTableForeignConstraintOidsInternal` etc ?

Reply:
You mean passing flags for filtering according to `PartitionMethod` too ?